Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -176,26 +176,31 @@ fileprivate enum TaskMetadata: DependencyTracker {
 
   init(_ notification: any NotificationType) {
     switch notification {
-    case is InitializedNotification:
-      self = .globalConfigurationChange
-    case is CancelRequestNotification:
-      self = .freestanding
-    case is ExitNotification:
-      self = .globalConfigurationChange
-    case let notification as DidOpenTextDocumentNotification:
-      self = .documentUpdate(notification.textDocument.uri)
-    case let notification as DidCloseTextDocumentNotification:
-      self = .documentUpdate(notification.textDocument.uri)
-    case let notification as DidChangeTextDocumentNotification:
-      self = .documentUpdate(notification.textDocument.uri)
-    case is DidChangeWorkspaceFoldersNotification:
-      self = .globalConfigurationChange
-    case is DidChangeWatchedFilesNotification:
-      self = .freestanding
-    case let notification as WillSaveTextDocumentNotification:
-      self = .documentUpdate(notification.textDocument.uri)
-    case let notification as DidSaveTextDocumentNotification:
-      self = .documentUpdate(notification.textDocument.uri)
+    case is CancelRequestNotification: self = .freestanding
+    case is CancelWorkDoneProgressNotification: self = .freestanding
+    case is DidChangeConfigurationNotification: self = .globalConfigurationChange
```

Comment:
Nit: IMO the switch is easier to read with the case bodies on a new line, but I don't feel too strongly about it.

Reply:
I don’t fee strongly about it either but had a slight preference for the single-line formatting. But since you prefer the multi-line and we need to abbreviate `notification` to make it fit on a single line, I changed it back to multi-line.